src/rsa/blinding.rs

```diff
@@ -0,0 +1,585 @@
+
+        let sampling_params = SamplingParams {
+            most_sig_limb_mask: Limb::max_value(),
+            reduce_when_over_bound: false,
+            extend_limbs_by_one: false,
+        };
+
+        let range = Range {
+            max_exclusive: &max_exclusive,
+            sampling_params: &sampling_params,
+        };
+
+        // Test that a generated zero is rejected and that `sample_into_limbs`
+        // gives up after a while of only getting zeros.
+        {
+            let mut result = [0, 0];
+            assert!(range.sample_into_limbs(&mut result, &random_00).is_err());
+        }
+
+        // Test that a generated value larger than `max_exclusive` is rejected
+        // and that `sample_into_limbs` gives up after a while of only getting
+        // values larger than the group order.
+        {
+            let mut result = [0, 0];
+            assert!(range.sample_into_limbs(&mut result, &random_ff).is_err());
+        }
+
+        // Test that a generated value exactly equal `max_exclusive` is
+        // rejected and that `generate` gives up after a while of only getting
+        // that value from the PRNG.
+        let max_exclusive_bytes = limbs_as_bytes(&max_exclusive);
+        {
+            let rng = rand::test_util::FixedSliceRandom {
+                bytes: &max_exclusive_bytes
+            };
+            let mut result = [0, 0];
+            assert!(range.sample_into_limbs(&mut result, &rng).is_err());
+        }
+
+        let max_exclusive_minus_1 = [max_exclusive[0] - 1, max_exclusive[1]];
+
+        // Test that a generated value exactly equal to `mex_exclusive - 1` is
+        // accepted.
+        let max_exclusive_minus_1_bytes =
+            limbs_as_bytes(&max_exclusive_minus_1);
+        {
+            let rng = rand::test_util::FixedSliceRandom {
+                bytes: max_exclusive_minus_1_bytes
+            };
+            let mut result = [0, 0];
+            range.sample_into_limbs(&mut result, &rng).unwrap();
+            assert_eq!(&max_exclusive_minus_1, &result);
+        }
+
+        // Test recovery from initial RNG failure.
+        {
+            let bytes = [
+                &max_exclusive_bytes[..],
+                &[0u8; 2 * LIMB_BYTES],
+                &max_exclusive_minus_1_bytes[..],
+            ];
+            let rng = rand::test_util::FixedSliceSequenceRandom {
+                bytes: &bytes,
+                current: core::cell::UnsafeCell::new(0),
+            };
+            let mut result = [0, 0];
+            range.sample_into_limbs(&mut result, &rng).unwrap();
+            assert_eq!(&max_exclusive_minus_1, &result);
+        }
+    }
+}
+
+#[cfg(feature = "internal_benches")]
+mod bench {
+    use {bench, rand};
+    use limb::*;
+    use super::{RSA_KEY_MAX_LIMBS, Range, SamplingParams, with_extended_buffers};
+
+    #[bench]
+    fn bench_sample_into_limbs_simple(b: &mut bench::Bencher) {
```

Comment:
Some comments about the purpose of this benchmark would be useful. In particular, why does the benchmark fill in its own `SamplingParams` and `Range` instead of having the real library code calculate them? What is meant by "simple" in the function name?